app/experimenter/static/js/components/TimelinePopForm.js

```diff
@@ -0,0 +1,372 @@
+      return (
+        <Container>
+          <div className="fa-5x" data-testId="spinner">
+            <Row className="justify-content-center">
+              <i className="fas fa-spinner fa-spin" />
+            </Row>
+          </div>
+        </Container>
+      );
+    }
+    return (
+      <div>
+        <Container>
+          <form onSubmit={this.handleSubmit}>
+            <Row className="mb-3 mt-3">
+              <h4 className="col-10 offset-2">Delivery Timeline</h4>
+            </Row>
+            <DesignInput
+              type="date"
+              label="Proposed Start Date"
+              name="proposed_start_date"
+              id="id_proposed_start_date"
+              onChange={value => {
+                this.handleDataChange("proposed_start_date", value);
+              }}
+              value={this.state.data.get("proposed_start_date")}
+              error={this.state.errors.get("proposed_start_date", "")}
+              helpContent={PROPOSED_START_DATE_HELP}
+              labelColumnWidth={2}
+            />
+            <DesignInput
+              label="Proposed Total Duration (days)"
+              name="proposed_duration"
+              id="id_proposed_duration"
+              onChange={value => {
+                this.handleDataChange("proposed_duration", value);
+              }}
+              value={this.state.data.get("proposed_duration")}
+              error={this.state.errors.get("proposed_duration", "")}
+              helpContent={PROPOSED_DURATION_HELP}
+              labelColumnWidth={2}
+            />
+            <DesignInput
+              label="Proposed Enrollment Duration (days)"
+              name="proposed_enrollment"
+              id="id_proposed_enrollment"
+              onChange={value => {
+                this.handleDataChange("proposed_enrollment", value);
+              }}
+              value={this.state.data.get("proposed_enrollment")}
+              error={this.state.errors.get("proposed_enrollment", "")}
+              helpContent={PROPOSED_ENROLLMENT_HELP}
+              labelColumnWidth={2}
+              optional={true}
+            />
+            <hr className="heavy-line my-5" />
+            <Row className="mb-3 mt-3">
+              <h4 className="col-10 offset-2">Delivery Population</h4>
+            </Row>
+            {this.displayPopulationPercent()}
+            <DesignInput
+              as="select"
+              label="Firefox Channel"
+              name="firefox_channel"
+              id="id_firefox_channel"
+              onChange={value => {
+                this.handleDataChange("firefox_channel", value);
+              }}
+              value={this.state.data.get("firefox_channel")}
+              error={this.state.errors.get("firefox_channel", "")}
+              helpContent={CHANNEL_HELP}
+              labelColumnWidth={2}
+              helpIsExternalLink={true}
+            >
+              <option>Firefox Channel</option>
```

Comment:
Is there a way to mark this option as disabled?  I realized recently we had a similar issue on the design page where you can set these back to the placeholder label and it'll save.

Reply:
Disabling that option could work, but isn't it OK that it saves when someone sets it back to Firefox Channel? Don't we want someone to be able to decide later that they want to leave that blank for a while? We just want to make sure that this part of the form isn't considered done if the value of channel is "Firefox Channel", right? Maybe the fix is to make sure that the model method that checks for the population form doneness checks that the value isn't "Firefox Channel"?

I can make it so that "Firefox Channel" isn't clickable, but again the end result of that will be once someone chooses a channel, they won't ever be able to set it back to what is essentailly supposed to be "blank" again.

Reply:
Yeah I see your point.  I don't think there's a case where we'd want someone to be able to set it back to blank.  Like they have to choose one, so they could go back and set it to another channel.

But it's also different on this page than design, because on design you have to fill everything out before you can save whereas here you can't, so if we'd have to make it so that the option is disabled (ie not selectable after they choose a value) but can still save if they havne't chosen anything yet, and those two things might conflict?

In which case the solution is probably to leave it selectable and do the thing we talked about elsewhere which is just highlight it as an incomplete field if they come back to the page.

In either case don't worry about it here, just leave it as is and we can find a better solution in a separate PR.  Thanks for thinking it through with me :)

Reply:
ok sounds good.